component/authenticator/src/main/java/org/wso2/carbon/identity/policy/password/PasswordResetEnforcer.java

```diff
@@ -0,0 +1,362 @@
+                                           AuthenticationContext context)
+            throws AuthenticationFailedException {
+        // if the logout request comes, then no need to go through and doing complete the flow.
+        if (context.isLogoutRequest()) {
+            return AuthenticatorFlowStatus.SUCCESS_COMPLETED;
+        }
+        if (StringUtils.isNotEmpty(request.getParameter(PasswordPolicyConstants.CURRENT_PWD))
+                && StringUtils.isNotEmpty(request.getParameter(PasswordPolicyConstants.NEW_PWD))
+                && StringUtils.isNotEmpty(request.getParameter(PasswordPolicyConstants.NEW_PWD_CONFIRMATION))) {
+            try {
+                processAuthenticationResponse(request, response, context);
+            } catch (Exception e) {
+                context.setRetrying(true);
+                context.setCurrentAuthenticator(getName());
+                return initiateAuthRequest(response, context, e.getMessage());
+            }
+            return AuthenticatorFlowStatus.SUCCESS_COMPLETED;
+        } else {
+            return initiateAuthRequest(response, context, null);
+        }
+    }
+
+    /**
+     * this will prompt user to change the credentials only if the last password
+     * changed time has gone beyond the pre-configured value.
+     *
+     * @param response the response
+     * @param context  the authentication context
+     */
+    private AuthenticatorFlowStatus initiateAuthRequest(HttpServletResponse response, AuthenticationContext context,
+                                                        String errorMessage)
+            throws AuthenticationFailedException {
+        // Find the authenticated user.
+        AuthenticatedUser authenticatedUser = getUser(context);
+        StepConfig stepConfig = context.getSequenceConfig().getStepMap().get(context.getCurrentStep() - 1);
+
+        if (authenticatedUser == null) {
+            throw new AuthenticationFailedException("Authentication failed!. " +
+                    "Cannot proceed further without identifying the user");
+        }
+
+        // The password reset flow for local authenticator
+        if (stepConfig.getAuthenticatedAutenticator().getApplicationAuthenticator() instanceof
+                LocalApplicationAuthenticator) {
+            String tenantDomain = authenticatedUser.getTenantDomain();
+            String username = authenticatedUser.getAuthenticatedSubjectIdentifier();
+            String tenantAwareUsername = MultitenantUtils.getTenantAwareUsername(username);
+
+            if (hadPasswordExpired(tenantDomain, tenantAwareUsername)) {
+                // The password has expired or the password changed time is not set
+                try {
+                    // Creating the URL to which the user will be redirected
+                    String loginPage = ConfigurationFacade.getInstance().getAuthenticationEndpointURL()
+                            .replace("login.do", "pwd-reset.jsp");
```

Comment:
define a constant and use it for "login.do", "pwd-reset.jsp"

Reply:
Added.